app/models/raw_input_changes.rb

```diff
@@ -41,13 +41,22 @@ def matching_organization
   end
 
   def matching_website
-    @matching_website ||= Website.find_by(content: url)
+    @matching_website ||= find_website(url) || find_website(resolved_url)
+  end
+
+  def find_website(content)
+    Website.find_by(content: content)
   end
 
   def url
     @attrs.fetch(:website, {})[:content]
   end
 
+  def resolved_url
+    @resolved_url ||= WebsiteResolver.resolve url
+    @attrs[:website][:content] = @resolved_url if @resolved_url
```

Comment:
I hate side-effects and this is one - how would you know that by calling the `resolved_url` method you'd also be mutating the `@attrs`?? But, I just couldn't think of a way around it - open to ideas!